### PR TITLE
Zap 66 pin

### PR DESCRIPTION
**Pull request by Beafowl-Pull**
## Describe your changes
- Add the ```pin``` command
- Fix tiles and inventory variables
## Issue ticket number and link
https://uwuclub.atlassian.net/browse/ZAP-66?atlOrigin=eyJpIjoiNWM4ZDI5MzljYjM4NDFiN2I1NWMwNzA1ZDRkOGEzYTEiLCJwIjoiaiJ9
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have documented my code
- [x] Do we need to implement unit tests?

### CODE REVIEW

File: src/gui/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ add_executable(zappy_gui
         cli/ClientApi.cpp
         cli/ParserData.hpp
         cli/ParserData.cpp
-        cli/ServerData.cpp
```
Comment (Valegox) [inline src/gui/CMakeLists.txt:9]: you remove `ServerData.cpp` ?

File: src/gui/cli/ClientApi.cpp [hunk 2]
```diff
@@ -225,13 +257,13 @@ namespace Zappy::GUI {
...
-
-            _serverData._timeUnit = std::stoi(myTime);
-        }
+    void ClientApi::ReceiveSgt(const std::string &aResponse)
+    {
+        std::string const &myArg = aResponse;
+        std::string const myTime = myArg.substr(0, myArg.find(' '));
 
-    } // namespace Zappy::GUI
+        _serverData._timeUnit = std::stoi(myTime);
```
Comment (Valegox) [inline src/gui/cli/ClientApi.cpp:267]: can you rename `_timeUnit` by `_freq` ?

File: src/gui/cli/Player.cpp [hunk 3]
```diff
@@ -5,8 +5,8 @@ namespace Zappy::GUI {
 
     Player::Player()
     {
+        std::cout << "put02" << std::endl;
```
Comment (Valegox) [inline src/gui/cli/Player.cpp:8]: debug message ?

File: src/gui/cli/Player.cpp [hunk 4]
```diff
@@ -34,12 +34,10 @@ namespace Zappy::GUI {
...
     }
 
-    void Player::setInventory(int aItem, long unsigned int aSlot)
+    void Player::setInventory(ItemPacket &aInventory)
     {
-        if (aSlot > 6) {
-            return;
-        }
-        _inventory[aSlot] = aItem;
+        std::cout << "put1" << std::endl;
```
Comment (Valegox) [inline src/gui/cli/Player.cpp:39]: debug message ?

File: src/gui/cli/Player.cpp [hunk 5]
```diff
@@ -64,10 +62,29 @@ namespace Zappy::GUI {
 
     int Player::getInventory(long unsigned int aSlot) const
     {
-        if (aSlot > 6) {
-            return -1;
+        switch (aSlot) {
```
Comment (Valegox) [inline src/gui/cli/Player.cpp:65]: you can avoid switch by creating an array containing all items. then you can return `array[aSlot]`

File: src/gui/main.cpp [hunk 6]
```diff
@@ -16,6 +14,8 @@ int main(int argc, char **argv)
             if (myClientApi.update() >= 1) {
                 break;
             }
+            myClientApi.update();
+            myClientApi.sendCommand("pin 4");
```
Comment (Valegox) [inline src/gui/main.cpp:18]: debug command call ?

**Review by Valegox — CHANGES_REQUESTED** [review]

File: src/gui/CMakeLists.txt (hunk 1 above)
Comment (Beafowl-Pull) [reply src/gui/CMakeLists.txt:9]: yeah, implementation are directly in hpp

File: src/gui/cli/Player.cpp (hunk 4 above)
Comment (Beafowl-Pull) [reply src/gui/cli/Player.cpp:39]: yeap sorry

File: src/gui/main.cpp (hunk 6 above)
Comment (Beafowl-Pull) [reply src/gui/main.cpp:18]: yeap sorry

**Review by Valegox — APPROVED** [review]

**Pull request merged**